src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/OrdinaryMethodReferenceFinder.cs

```diff
@@ -129,22 +143,32 @@ private bool IsGetAwaiterMethod(IMethodSymbol methodSymbol)
 
             if (IsForEachMethod(symbol))
             {
-                var forEachMatches = await FindReferencesInForEachStatementsAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+                var forEachMatches = await FindReferencesInForEachStatementsAsync(
+                    symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
                 nameMatches = nameMatches.Concat(forEachMatches);
             }
 
             if (IsDeconstructMethod(symbol))
             {
-                var deconstructMatches = await FindReferencesInDeconstructionAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+                var deconstructMatches = await FindReferencesInDeconstructionAsync(
+                    symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
                 nameMatches = nameMatches.Concat(deconstructMatches);
             }
 
             if (IsGetAwaiterMethod(symbol))
             {
-                var getAwaiterMatches = await FindReferencesInAwaitExpressionAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+                var getAwaiterMatches = await FindReferencesInAwaitExpressionAsync(
+                    symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
                 nameMatches = nameMatches.Concat(getAwaiterMatches);
             }
 
+            if (IsSliceMethod(symbol) || IsSubstringMethod(symbol) || IsSubArrayMethod(symbol))
```

Comment:
should we check the parent type as well. for example, this needs to be String.Substring right, not any method called Substring.